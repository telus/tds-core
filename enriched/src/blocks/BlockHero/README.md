Example:

    <BlockHero 
        title="This is the title" 
        eyebrow="Some small text above the title"
        heroImage={{ file: { url: 'http://via.placeholder.com/1440x400' }}}
    />
