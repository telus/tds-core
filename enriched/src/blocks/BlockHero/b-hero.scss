@import '~telus-thorium-core/scss/settings/variables';

.b-hero {
  &__image:after {
    padding-bottom: 8.13037%;
    background-image: url(//cdn.business.telus.digital/v2-assets/images/b4yb/waves/wave_landing.png);
    background-position: center bottom;
    background-repeat: no-repeat;
    background-size: 100%, auto, contain;
    bottom: -1px;
    content: '';
    left: 0;
    position: absolute;
    right: 0;
  }
}
