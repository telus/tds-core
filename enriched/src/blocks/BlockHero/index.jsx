import PropTypes from 'prop-types';
import React from 'react';

if (process.env.BROWSER) {
  require('./b-hero.scss');
}

/**
 * BlockHero is usually rendered at the top of a page. It consists of a title and an eyebrow (left-aligned),
 * and a hero image as the background overlaid with a wave image.
 * Other variations include [BlockHeroOverlay](#blockherooverlay)
 *
 * Contentful models (BUS > Marketing): Headline Block
 */
const BlockHero = ({ eyebrow, title, heroImage }) => {
  const headerImage = {
    backgroundImage: `url(${heroImage && heroImage.file && heroImage.file.url})`,
    paddingBottom: '28.125%',
    backgroundPosition: 'center top',
    backgroundSize: 'cover',
    position: 'relative',
    width: '100%'
  };

  return (
    <section className="b-hero">
      <div className="b-hero__image" style={headerImage} />
      <div className="container container--limited-width">
        <div className="grid-row">
          <div className="medium-12">
            <p className="heading-4 b-hero__eyebrow">{eyebrow}</p>
            <h1 className="display-heading-1">{title}</h1>
          </div>
        </div>
      </div>
    </section>
  );
};

BlockHero.propTypes = {
  /** recommended: text should fit in one line */
  title: PropTypes.string.isRequired,
  /** recommended: text should fit in one line */
  eyebrow: PropTypes.string,
  /** Contentful media field, recommended specification: greater than 1440 x 400 */
  heroImage: PropTypes.shape({
    file: PropTypes.shape({
      url: PropTypes.string.isRequired
    }).isRequired
  })
};

BlockHero.defaultProps = {
  eyebrow: ''
};

export default BlockHero;
