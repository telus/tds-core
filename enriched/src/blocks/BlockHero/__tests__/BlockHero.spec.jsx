import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import BlockHero from '../';

describe('BlockHeroTitleEyebrow', () => {
  const stubHeroImage = { file: { url: 'some/url', title: 'Client logo title' } };
  const data = { eyebrow: 'Foo', title: 'Bar', heroImage: stubHeroImage };

  it('matches snapshot', () => {
    const wrapper = mount(<BlockHero {...data} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render eyebrow', () => {
    const wrapper = mount(<BlockHero {...data} />);
    expect(wrapper.find('p').first().render().text()).toEqual('Foo');
  });

  it('should render without subcategory', () => {
    const wrapper = mount(<BlockHero title={data.title} heroImage={stubHeroImage} />);
    expect(wrapper.find('p').first().render().text()).toEqual('');
  });

  it('should render title', () => {
    const wrapper = mount(<BlockHero {...data} />);
    expect(wrapper.find('h1').first().render().text()).toEqual('Bar');
  });

  it('should use headline style', () => {
    const wrapper = mount(<BlockHero {...data} />);
    expect(wrapper.find('section').children().first().hasClass('b-hero__image')).toBeTruthy();
  });

  it('should add the heroimage as a background', () => {
    const wrapper = mount(<BlockHero {...data} />);
    expect(wrapper.find('.b-hero__image').prop('style')).toEqual({ backgroundImage: 'url(some/url)', backgroundPosition: 'center top', backgroundSize: 'cover', paddingBottom: '28.125%', position: 'relative', width: '100%' });
  });
});
