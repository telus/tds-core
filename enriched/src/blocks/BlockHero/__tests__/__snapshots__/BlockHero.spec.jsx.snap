// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockHeroTitleEyebrow matches snapshot 1`] = `
<BlockHero
  eyebrow="Foo"
  heroImage={
    Object {
      "file": Object {
        "title": "Client logo title",
        "url": "some/url",
      },
    }
  }
  title="Bar"
>
  <section
    className="b-hero"
  >
    <div
      className="b-hero__image"
      style={
        Object {
          "backgroundImage": "url(some/url)",
          "backgroundPosition": "center top",
          "backgroundSize": "cover",
          "paddingBottom": "28.125%",
          "position": "relative",
          "width": "100%",
        }
      }
    />
    <div
      className="container container--limited-width"
    >
      <div
        className="grid-row"
      >
        <div
          className="medium-12"
        >
          <p
            className="heading-4 b-hero__eyebrow"
          >
            Foo
          </p>
          <h1
            className="display-heading-1"
          >
            Bar
          </h1>
        </div>
      </div>
    </div>
  </section>
</BlockHero>
`;
